File: 2021/src/test/java/de/jfriemel/aoc/days/Day17Test.java
package de.jfriemel.aoc.days;

import org.junit.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class Day17Test {

    private static final String INPUT_STRING = "target area: x=20..30, y=-10..-5";

    private static final List<String> INPUT = Arrays.asList(INPUT_STRING.split("\n"));

    @Test
    public void testPart1() {
        String result = new Day17().part1(INPUT);
        assertEquals("45", result);
    }

    @Test
    public void testPart2() {
        String result = new Day17().part2(INPUT);
        assertEquals("112", result);
    }

}

File: 2023/src/test/kotlin/de/jfriemel/aoc/days/Day17Test.kt
package de.jfriemel.aoc.days

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class Day17Test {
    private val input1 = """
        2413432311323
        3215453535623
        3255245654254
        3446585845452
        4546657867536
        1438598798454
        4457876987766
        3637877979653
        4654967986887
        4564679986453
        1224686865563
        2546548887735
        4322674655533
    """.trimIndent().lines()

    private val input2 = """
        111111111111
        999999999991
        999999999991
        999999999991
        999999999991
    """.trimIndent().lines()

    @Test
    fun testPart1() {
        val result = Day17.part1(input1)
        Assertions.assertEquals("102", result)
    }

    @Test
    fun testPart2_1() {
        val result = Day17.part2(input1)
        Assertions.assertEquals("94", result)
    }

    @Test
    fun testPart2_2() {
        val result = Day17.part2(input2)
        Assertions.assertEquals("71", result)
    }
}
